bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -91,7 +91,29 @@ public LinkyHandler(Thing thing, LocaleProvider localeProvider, Gson gson, HttpC
 
         this.cachedDailyData = new ExpiringDayCache<>("daily cache", REFRESH_FIRST_HOUR_OF_DAY, () -> {
             LocalDate today = LocalDate.now();
-            return getConsumptionData(today.minusDays(15), today);
+            Consumption consumption = getConsumptionData(today.minusDays(15), today);
+            if (consumption != null) {
+                Aggregate days = consumption.aggregats.days;
+                Aggregate weeks = consumption.aggregats.weeks;
+                if (logger.isDebugEnabled()) {
+                    for (int i = 0; i < days.datas.size(); i++) {
+                        logger.debug("Day {} value {}",
+                                days.periodes.get(i).dateDebut.format(DateTimeFormatter.ISO_LOCAL_DATE),
+                                days.datas.get(i));
+                    }
+                    for (int i = 0; i < weeks.datas.size(); i++) {
+                        logger.debug("Weeek {} {} value {}",
+                                weeks.periodes.get(i).dateDebut.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME),
+                                weeks.periodes.get(i).dateFin.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME),
+                                weeks.datas.get(i));
+                    }
+                }
+                if (days.datas.get(days.datas.size() - 1).isNaN()) {
```

Comment:
You should make sure that `days.datas` isn't empty here.

Reply:
Done